tortoise/src/main/scala/ProcedureCompiler.scala

```diff
@@ -18,8 +18,9 @@ class ProcedureCompiler(handlers: Handlers)(implicit compilerFlags: CompilerFlag
   def compileProcedures(procedureDefs: Seq[ProcedureDefinition]): CompiledProceduresDictionary =
     procedureDefs.map(compileProcedureDef)
 
-  private def compileProcedureDef(pd:            ProcedureDefinition)
+  private def compileProcedureDef(originalPd:    ProcedureDefinition)
                         (implicit compilerFlags: CompilerFlags, compilerContext: CompilerContext): (String, Seq[String]) = {
+    val pd = optimize.Optimizer(originalPd)
```

Comment:
The only reason I care about the flag is for testing purposes in the JVM.  We don't need it in the browser.

The field in `CompilerFlags` is a fine solution to me.